fixtures/provider.py

```diff
@@ -67,8 +76,27 @@ def _artifactor_skip_providers(request, providers, skip_msg):
     pytest.skip(skip_msg)
 
 
-def _setup_provider_verbose(provider):
+def _setup_provider_verbose(request, provider):
     try:
+        if request.config.option.provider_limit > 0:
+            # TODO: Use appliance.
+            existing_providers = [
+                p for p in list_providers(use_global_filters=False)
+                if p.exists and p.key != provider.key]
```

Comment:
**OPTIONAL** You dont need to compare provider keys to check for equality. You can compare them directly `...and p != provider]` since `BaseProvider` has `__eq__` defined.